frontends/p4/coreLibrary.cpp

```diff
@@ -15,9 +15,19 @@ limitations under the License.
 */
 
 #include "coreLibrary.h"
+#include "fromv1.0/v1model.h"
 
 namespace P4 {
 
 P4CoreLibrary P4CoreLibrary::instance;
 
 }  // namespace P4
+
+/* These must be in the same compiliation unit to ensure that P4CoreLibrary::instance
+ * is initialized before V1Model::instance */
+namespace P4V1 {
```

Comment:
The name of this cpp file should be changed.
It should perhaps be called "modelInstances.cpp"